packages/next/src/server/lib/router-server.ts

```diff
@@ -574,6 +532,53 @@ export async function initialize(opts: {
   }
   requestHandlers[opts.dir] = requestHandler
 
+  const renderServerOpts: Parameters<RenderServer['initialize']>[0] = {
+    port: opts.port,
+    dir: opts.dir,
+    hostname: opts.hostname,
+    minimalMode: opts.minimalMode,
+    dev: !!opts.dev,
+    server: opts.server,
+    isNodeDebugging: !!opts.isNodeDebugging,
+    serverFields: developmentBundler?.serverFields || {},
+    experimentalTestProxy: !!opts.experimentalTestProxy,
+    experimentalHttpsServer: !!opts.experimentalHttpsServer,
+    bundlerService: devBundlerService,
+    startServerSpan: opts.startServerSpan,
+  }
+
+  if (!renderServerOpts.serverFields) {
+    renderServerOpts.serverFields = {}
+  }
```

Comment:
This `if` statement will always be false since line 543 makes sure its an object instead of null/undefined